mii/method_table.py

```diff
@@ -140,19 +142,162 @@ def text2img_pack_response_to_proto(response, time_taken, model_time_taken):
+
+    def unpack_response_from_proto(self, response):
+        return response
+
+
+class TextGenerationMethods(TaskMethods):
+    session_context = {}
+
+    @property
+    def method(self):
+        return "GeneratorReply"
+
+    pack_request_to_proto = multi_string_request_to_proto
+    unpack_request_from_proto = proto_request_to_list
+
+    def create_session(self, session_id):
+        if session_id in self.session_context:
+            raise ValueError(f"session {session_id} already exists")
+        self.session_context[session_id] = None
+
+    def destroy_session(self, session_id):
+        if session_id not in self.session_context:
+            raise ValueError(f"session {session_id} does not exist")
+        del self.session_context[session_id]
+
+    def preprocess_session(self, session_id, args):
+        if session_id not in self.session_context:
+            raise ValueError(f"session {session_id} does not exist")
+        if self.session_context[session_id] is None:
+            self.session_context[session_id] = ""
+        if len(args[0]) != 1:
+            raise ValueError(f"You can pass only one prompt with a session_id")
+
+        args = ([self.session_context[session_id] + args[0][0]], )
+        return args
+
+    def run_inference(self, inference_pipeline, args, kwargs):
+        session_id = kwargs.pop("session_id", None)
+        if session_id:
+            args = self.preprocess_session(session_id, args)
+        response = inference_pipeline(*args, **kwargs)
+
+        if session_id:
+            response = self.postprocess_session(session_id, args, response)
+
+        return response
+
+    def postprocess_session(self, session_id, args, response):
+        generated_text = response[0][0]["generated_text"]
+        self.session_context[session_id] = generated_text
+        response[0][0]["generated_text"] = generated_text[len(args[0][0]):]
+        return response
+
+    def pack_response_to_proto(self, response, time_taken, model_time_taken):
+        text_responses = []
+        for response in response:
+            text = response[0]['generated_text']
+            text_responses.append(text)
+
+        return modelresponse_pb2.MultiStringReply(response=text_responses,
+                                                  time_taken=time_taken,
+                                                  model_time_taken=model_time_taken)
+
+
+class TextClassificationMethods(TaskMethods):
+    @property
+    def method(self):
+        return "ClassificationReply"
+
+    pack_request_to_proto = single_string_request_to_proto
+    unpack_request_from_proto = proto_request_to_single_input
+    pack_response_to_proto = single_string_response_to_proto
+
+
+class QuestionAnsweringMethods(TaskMethods):
+    @property
+    def method(self):
+        return "QuestionAndAnswerReply"
+
+    pack_request_to_proto = question_answering_pack_request_to_proto
+    unpack_request_from_proto = question_answering_unpack_request_from_proto
```

Comment:
Since these functions are specific to `QuestionAnsweringMethods` and not used in the other tasks, can we bring the definitions into the class itself? Same for the other task-specific functions